pkg/apis/pipeline/v1alpha1/build_gcs_resource.go

```diff
@@ -0,0 +1,155 @@
+	GCSManifest GCSArtifactType = "Manifest"
+)
+
+// BuildGCSResource describes a resource in the form of an archive,
+// or a source manifest describing files to fetch.
+type BuildGCSResource struct {
+	Name           string
+	Type           PipelineResourceType
+	Location       string
+	DestinationDir string
+	ArtifactType   GCSArtifactType
+	//Secret holds a struct to indicate a field name and corresponding secret name to populate it
+	Secrets []SecretParam `json:"secrets"`
+}
+
+// NewBuildGCSResource creates a new BuildGCS resource to pass to knative build
+func NewBuildGCSResource(r *PipelineResource) (*BuildGCSResource, error) {
+	if r.Spec.Type != PipelineResourceTypeStorage {
+		return nil, fmt.Errorf("BuildGCSResource: Cannot create a BuildGCS resource from a %s Pipeline Resource", r.Spec.Type)
+	}
+	var location, destDir string
+	var aType GCSArtifactType
+	var locationSpecified, aTypeSpecified bool
+
+	for _, param := range r.Spec.Params {
+		switch {
+		case strings.EqualFold(param.Name, "Location"):
+			location = param.Value
+			if param.Value != "" {
+				locationSpecified = true
+			}
+		case strings.EqualFold(param.Name, "DestinationDir"):
+			destDir = param.Value
+		case strings.EqualFold(param.Name, "ArtifactType"):
+			aType = GCSArtifactType(param.Value)
+			if param.Value != "" {
+				aTypeSpecified = true
+			}
+		}
+	}
+
+	if !locationSpecified {
+		return nil, fmt.Errorf("BuildGCSResource: Need Location to be specified in order to create BuildGCS resource %s", r.Name)
+	}
+	if !aTypeSpecified {
+		return nil, fmt.Errorf("BuildGCSResource: Need ArtifactType to be specified in order to fetch BuildGCS resource %s", r.Name)
+	}
+	return &BuildGCSResource{
+		Name:           r.Name,
+		Type:           r.Spec.Type,
+		Location:       location,
+		DestinationDir: destDir,
+		ArtifactType:   aType,
+		Secrets:        r.Spec.SecretParams,
+	}, nil
+}
+
+// GetName returns the name of the resource
+func (s BuildGCSResource) GetName() string {
+	return s.Name
+}
+
+// GetType returns the type of the resource, in this case "storage"
+func (s BuildGCSResource) GetType() PipelineResourceType {
+	return PipelineResourceTypeStorage
+}
+
+// GetParams get params
+func (s *BuildGCSResource) GetParams() []Param { return []Param{} }
+
+// GetSecretParams returns the resource secret params
+func (s *BuildGCSResource) GetSecretParams() []SecretParam { return s.Secrets }
+
+// GetDownloadContainerSpec returns an array of container specs to download gcs storage object
+func (s *BuildGCSResource) GetDownloadContainerSpec() ([]corev1.Container, error) {
+	if s.DestinationDir == "" {
+		return nil, fmt.Errorf("BuildGCSResource: Expect Destination Directory param to be set %s", s.Name)
+	}
+	args := []string{"--type", string(s.ArtifactType), "--location", s.Location}
+	// dest_dir is the destination directory for GCS files to be copies"
+	if s.DestinationDir != "" {
+		args = append(args, "--dest_dir", filepath.Join(workspaceDir, s.DestinationDir))
```

Comment:
Is there any need to prefix workspaceDir here? 
I think we should safely rely on DestinationDir providing full path.

Reply:
yes. i think we can do that.